src/Core/Framework/Api/OAuth/ScopeRepository.php

```diff
@@ -68,7 +69,7 @@ public function finalizeScopes(
             $hasWrite = true;
         }
 
-        if (!$hasWrite) {
+        if (!$hasWrite && $grantType !== 'refresh_token') {
```

Comment:
would make sense to move the supported grant types into an ENUM, not very relevant for this PR though..

Reply:
Sadly this the lib itself does not provide this as enum or anything (not even consts) -> I'm not sure if it makes sense for us to be more strict here if the lib does not provide it 🤔 

As an improvement I add some consts for now